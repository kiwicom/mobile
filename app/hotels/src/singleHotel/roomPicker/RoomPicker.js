// @flow strict

import * as React from 'react';

import SelectButton from './SelectButton';
import IncrementDecrement from './IncrementDecrement';

type Price = {|
  +amount: number | null,
  +currency: string | null,
|};

type Props = {|
  +price: Price | null,
  +selectedCount: number,
  +selectableCount: number,
  +increment: () => void,
  +decrement: () => void,
|};

export default function RoomPicker(props: Props) {
  if (props.price == null) {
    return null;
  }

  if (props.selectedCount === 0) {
    return <SelectButton price={props.price} increment={props.increment} />;
  }

  return <IncrementDecrement {...props} />;
}
