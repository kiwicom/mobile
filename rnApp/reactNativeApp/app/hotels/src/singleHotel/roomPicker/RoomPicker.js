// @flow

import * as React from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
import { Price } from '@kiwicom/react-native-app-common';

import NoneSelected from './NoneSelected';

const styles = StyleSheet.create({
  container: {
    borderRadius: 3,
    backgroundColor: 'rgba(0, 151, 169, 0.2)',
    flexDirection: 'row',
  },
  button: {
    backgroundColor: '#0097a9',
    width: 60,
    paddingVertical: 10,
    alignItems: 'center',
  },
  minusButton: {
    borderTopLeftRadius: 3,
    borderBottomLeftRadius: 3,
  },
  plusButton: {
    borderTopRightRadius: 3,
    borderBottomRightRadius: 3,
  },
  disabledButton: {
    opacity: 0.2,
  },
  buttonText: {
    backgroundColor: 'transparent',
    fontSize: 20,
    fontWeight: '600',
    color: '#ffffff',
  },
  priceAndCount: {
    flexGrow: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  count: {
    fontSize: 14,
    color: '#0097a9',
    fontWeight: '800',
  },
  price: {
    fontSize: 14,
    color: '#0097a9',
  },
});

type Props = {|
  price: number,
  currency: string,
  selectedCount: number,
  selectableCount: number,
  increment: () => void,
  decrement: () => void,
|};

export default function RoomPicker({
  price,
  currency,
  selectedCount,
  selectableCount,
  increment,
  decrement,
}: Props) {
  if (selectedCount === 0) {
    return (
      <NoneSelected price={price} currency={currency} onSelect={increment} />
    );
  }

  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={[styles.button, styles.minusButton]}
        onPress={decrement}
      >
        <Text style={styles.buttonText}>-</Text>
      </TouchableOpacity>
      <View style={styles.priceAndCount}>
        <Text style={styles.count}>{selectedCount} Ã— </Text>
        <Price amount={price} currency={currency} style={styles.price} />
      </View>
      {selectedCount < selectableCount ? (
        <TouchableOpacity
          style={[styles.button, styles.plusButton]}
          onPress={increment}
        >
          <Text style={styles.buttonText}>+</Text>
        </TouchableOpacity>
      ) : (
        <View style={[styles.button, styles.plusButton, styles.disabledButton]}>
          <Text style={styles.buttonText}>+</Text>
        </View>
      )}
    </View>
  );
}
