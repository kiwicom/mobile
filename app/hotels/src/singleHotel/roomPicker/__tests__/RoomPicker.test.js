// @flow

import * as React from 'react';
import { PlaygroundRenderer } from '@kiwicom/mobile-playground';

import RoomPicker from '../RoomPicker';

const VoidAction = () => {};

it('renders as expected', () => {
  PlaygroundRenderer.render(
    <RoomPicker
      price={{ amount: 100, currency: 'CZK' }}
      selectedCount={0}
      selectableCount={1}
      increment={VoidAction}
      decrement={VoidAction}
    />,
  );

  PlaygroundRenderer.render(
    <RoomPicker
      price={{ amount: 100, currency: 'CZK' }}
      selectedCount={1}
      selectableCount={2}
      increment={VoidAction}
      decrement={VoidAction}
    />,
  );

  PlaygroundRenderer.render(
    <RoomPicker
      price={{ amount: 100, currency: 'CZK' }}
      selectedCount={2}
      selectableCount={2}
      increment={VoidAction}
      decrement={VoidAction}
    />,
  );
});
