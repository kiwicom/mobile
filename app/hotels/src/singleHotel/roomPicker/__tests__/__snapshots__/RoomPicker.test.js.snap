// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<SelectButton
  increment={[Function]}
  price={
    Object {
      "amount": 100,
      "currency": "CZK",
    }
  }
/>
`;

exports[`renders as expected 2`] = `
<IncrementDecrement
  decrement={[Function]}
  increment={[Function]}
  price={
    Object {
      "amount": 100,
      "currency": "CZK",
    }
  }
  selectableCount={2}
  selectedCount={1}
/>
`;

exports[`renders as expected 3`] = `
<IncrementDecrement
  decrement={[Function]}
  increment={[Function]}
  price={
    Object {
      "amount": 100,
      "currency": "CZK",
    }
  }
  selectableCount={2}
  selectedCount={2}
/>
`;
